### ISSUE TITLE
Highscore list for mods using arbitrary formula in schema

### ISSUE THREAD

**Issue by jimmydorry** [GetDotaStats/site#184]
Raised by @K1llMan

- [ ] Implement the scoring mechanism allowing modders to set an arbitrary scoring function (a combination of player values with weights)
- [ ] Page to display top 20 players per mod
- [ ] JSON API that has top 20 players per mod
- [ ] Implementation example for game mod

**Comment by jimmydorry** [comment]
Old conversation screen shot, because it was in the wrong place:

http://i.imgur.com/6HdN9f4.png

**Comment by K1llMan** [comment]
What is about individual stats for player? I think in will be useful to have possibility to show current player high score. It's may be a player individual goal to bet his best score and to have knowing about increase of his game skill.

**Comment by jimmydorry** [comment]
Yep. We'll start with top 20. The logic can then be mostly re-used for individual top score.

**Comment by jimmydorry** [comment]
@cowsforyou said:

jimmydorry,

First and foremost, thank you for upkeeping such a great site for the modding community! However, useful as it is for us modders, it's currently not very useful for the vast majority who are the players. Your site features a nice ladder/ranking system for the mods AND your site also tracks the no. of games each player plays per mod. So a simple tweak would be to add another tab into each individual mod which shows the player rankings (ranked based on the no. of games they played for the last 7 trailing days and total - like how you have it currently). Ideally it should be by winrate but that would encourage losing hosts to simply dc - whereas if it only ranks according to the number of games then people will be more inclined to play till the end.

That alone should encourage more people to come visit your site to keep tabs on their ranking (because well, everyone has a big ego). BUT that's only if they stumble across it. So to help with a bit of marketing, we can go one step further to make that player ranking some kind of RSS feed where modders can input into their loading screen (I'm not sure if it's possible? but who knows?). And if we can't feed it live, we can update it manually every day, every week, however lazy or hardworking we wanna be. I personally have my loading screen updated with all the patches I push through so I can't imagine it being that much difficult to also update the every time I push an update.

Advantage for the modders: With people now aware of the ladder, it's likely that they will play more - which is what all the modders are hoping for anyway.

Advantage for GDS: You should get a healthy boost in visits because now the players will be checking the site and checking it pretty often. That increases the value of your website and makes it more attractive to advertisers if you so wish to place some on your site.

Future evolution of GDS: With your site now well known among both the players and the modders, you can evolve it to be a central hub for other things - maybe tournaments for custom games (which I have hosted on mine recently), or pretty much whatever you can conceive. This will again boost your traffic higher and again make your site even more valuable.

Bottom line: It should be a win-win-win situation for everyone.

Contingencies: 1) There will be hackarounds to gaming the system of course. There will be stat-whores who will wanna just up their games just because. An ez ban on the player can solve the trick. Or if you want to let the modders control that you can also give them the power to them to ban players for their own mod. 2) Moving towards a winrate ladder is almost ideal if you want to make it more competitive and though not possible for most of the custom games, it could be implemented for those with dedicated servers.

**Comment by cowsforyou** [comment]
@jimmydorry  My bad for posting on the wrong git. I'm fairly new to GDS :)

**Comment by jimmydorry** [comment]
@mnoya said:

Sort of a generic highscore board, showing Player / Num Games / Num Wins

**Comment by cowsforyou** [comment]
@jimmydorry  Yeah what Noya said is accurate. I told him about my thoughts and he suggested I ping you like crazy on IRC.

**Comment by jimmydorry** [comment]
Issues are far better and leave a record.

**Comment by cowsforyou** [comment]
Will leave issues from now onward then. I see it's on your weekend to-dos! Looking forward to it.
